File: leetcode/problems/most_frequent_even_element/native_solution.py
class Solution:
    def mostFrequentEven(self, nums):
        count = {}
        for num in nums:
            if num % 2 == 0:
                if num not in count:
                    count[num] = 1
                else:
                    count[num] += 1
        print(count)
        most_frequent_key = 0
        most_frequent_value = 0

        for key, value in count.items():
            if value > most_frequent_value:
                most_frequent_value = value
                most_frequent_key = key
            elif value == most_frequent_value:
                if key < most_frequent_key:
                    most_frequent_key = key
        if most_frequent_value == 0:
            return -1
        # если print(most_frequent_key),
        # функция вернет None
        else:
            return most_frequent_key


nums = [0, 1, 4, 4, 2, 2, 1]
# nums = [29, 47, 21, 41, 13, 37, 25, 7]
solution = Solution()
result = solution.mostFrequentEven(nums)
print(result)

File: other/javascript_tasks/happy_ticket/native_solution.js
const sumOfSquareDigits = (number) => {
    let result = 0;
    let count = 0;
    while (count !== 10) {
        for (let i=0; i<number.length; i++) {
            result += number[i] * number[i];
        }
        if (result === 1) {
            return true;
        }
        number = result.toString();
        count +=1;
        result = 0;
    }
    return false;
}

console.log(sumOfSquareDigits('6')); // false
console.log(sumOfSquareDigits('7')); // true

// 7   => 7^2 = 49,
// 49  => 4^2 + 9^2 = 16 + 81 = 97,
// 97  => 9^2 + 7^2 = 81 + 49 = 130,
// 130 => 1^2 + 3^2 + 0^2 = 10,
// 10  => 1^2 + 0^2 = 1.
